src/com/palmergames/bukkit/towny/object/Town.java

```diff
@@ -729,6 +726,58 @@ private void remove(Resident resident) {
 		TownyUniverse.getInstance().getDataSource().saveTown(this);
 	}
 
+	/**
+	 * Tries to find a new mayor from among the town's residents with the rank specified.
+	 * 
+	 * @param rank - the rank being checked for potential mayors
+	 * @return found - whether or not a new mayor was found
+	 */
+	private boolean findNewMayor(String rank) {
+		boolean found = false;
+		System.out.println("getRank(rank = " + rank + ") = " + getRank(rank));
+		for (Resident newMayor : new ArrayList<>(getRank(rank))) {
+			System.out.println("newMayor = " + newMayor);
+			if ((newMayor != mayor) && (newMayor.hasTownRank(rank))) {  // The latter portion seems redundant.
+				try {
+					setMayor(newMayor);
+					found = true;
+					break;
+				} catch (TownyException e) {
+					// Error setting mayor.
+					e.printStackTrace();
+					found = false;
+				}
+				System.out.println("mayor = " + mayor);
+			}
+		}
+		return found;
+	}
+	
+	/**
+	 * Tries to find a new mayor from among the town's residents.
+	 * 
+	 * @return found - whether or not a new mayor was found
+	 */
+	private boolean findNewMayor() {
+		boolean found = false;
+		for (Resident newMayor : new ArrayList<>(getResidents())) {
```

Comment:
Again another array copu

Reply:
This should also be resolved as of 9606f94d6c95ec81a934e87eca2e5ac72278463d.